### PR TITLE
Update __truediv__ comment in python bindings.

### PR DESCRIPTION
**Pull request by rdspring1**
This PR updates the comments for `__truediv__` operator defined in python bindings. The current comment does not reflect what the code actually does.

Reference: https://github.com/NVIDIA/Fuser/pull/2837#pullrequestreview-2282427318

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/python_bindings.cpp [hunk 1]
```diff
@@ -2056,14 +2056,19 @@ void initNvFuserPythonBindings(PyObject* module) {
...
-  // When applied to integer-dtype arguments, they do as expected, returning
-  // integer and float outputs, respectively. When applied to two floating-type
-  // arguments, they return the floor of division for // and plain division for
-  // /. When applied to mixed types, the types are promoted, so the
-  // floating-point behavior is returned.
-  // Our div operator matches the __truediv__ behavior, so we do not implement
-  // __div__.
+  // In python, __truediv__ (/) always returns a float regardless of whether
+  // the input arguments are float or integer. __truediv__ (/) corresponds with
+  // pytorch torch.true_divide(a, b). The __div__ operator is depreciated in
```
Comment (jacobhinkle) [inline csrc/python_frontend/python_bindings.cpp:2061]:
```suggestion
  // pytorch torch.true_divide(a, b). The __div__ operator is deprecated in
```

File: csrc/python_frontend/python_bindings.cpp [hunk 2]
```diff
@@ -2056,14 +2056,19 @@ void initNvFuserPythonBindings(PyObject* module) {
...
+  // python 3.
+  //
+  // In nvfuser, truediv function has the same semantics as python's
+  // operator __truediv__ (/). The div function truncates the result instead
+  // of promoting it to float, so it has the same semantics as the C++'s (/)
+  // operator. In pytorch, torch.div(a, b, rounding_mode='trunc') corresponds
+  // C-style integer division.
```
Comment (jacobhinkle) [inline csrc/python_frontend/python_bindings.cpp:2068]: The first sentence here is incorrect I think. On Python 3.12, `__truediv__` of ints promotes to double first and does no truncation:
```
[ins] In [1]: x = 3

[ins] In [2]: y = 4

[ins] In [3]: x / y
Out[3]: 0.75

[ins] In [4]: x.__truediv__(4)
Out[4]: 0.75
```
To get truncdiv in Python I think you need to manually truncate:
```
[ins] In [17]: ((-30)/4).__trunc__()
Out[17]: -7

[ins] In [18]: (-30)//4
Out[18]: -8
```

Comment (rdspring1) [reply csrc/python_frontend/python_bindings.cpp]: How is the first sentence incorrect? It is what the comment says in `csrc/ops/arith.h`

> // truediv: promote to float for integer division, has the same semantics as the
> // python's operator /

https://github.com/NVIDIA/Fuser/blob/main/csrc/ops/arith.h#L426-L431

> // div: don't promote to float, instead, truncate the result, this has the same
> // semantics as the C++'s operator /

https://github.com/NVIDIA/Fuser/blob/main/csrc/ops/arith.h#L432-L437

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/python_bindings.cpp (hunk 2 above)
Comment (rdspring1) [reply csrc/python_frontend/python_bindings.cpp]: I updated the comment to directly mention `csrc/ops/arith.h`.

We are changing the dunder `__truediv__` in our python API to map to c-style division.

File: csrc/python_frontend/python_bindings.cpp [hunk 3]
```diff
@@ -2058,14 +2058,15 @@ void initNvFuserPythonBindings(PyObject* module) {
...
-  // In nvfuser, truediv function has the same semantics as python's
-  // operator __truediv__ (/). The div function truncates the result instead
-  // of promoting it to float, so it has the same semantics as the C++'s (/)
-  // operator. In pytorch, torch.div(a, b, rounding_mode='trunc') corresponds
-  // C-style integer division.
+  // In nvfuser, truediv function in csrc/ops/arith.h has the same semantics as
+  // python's operator __truediv__ (/). The div function in csrc/ops/arith.h
+  // truncates the result instead of promoting it to float. It has the same
+  // semantics as the C++'s (/) operator. In pytorch,
+  // torch.div(a, b, rounding_mode='trunc') corresponds C-style integerw
```
Comment (jacobhinkle) [inline csrc/python_frontend/python_bindings.cpp:2068]:
```suggestion
  // torch.div(a, b, rounding_mode='trunc') corresponds C-style integer
```

**Review by jacobhinkle — APPROVED** [review]
LGTM!

File: csrc/python_frontend/python_bindings.cpp (hunk 3 above)
Comment (rdspring1) [reply csrc/python_frontend/python_bindings.cpp]: I blame vim. 😄

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Pull request merged**